### PR TITLE
 use BackpropType. enum

### PR DESCRIPTION
**Pull request by makosblade**
This PR:
- Updates socrates to use updated APIs

### CODE REVIEW

**Review by katherinebernstein — APPROVED** [review]

File: src/main/java/com/openlattice/socrates/Socrates.java [hunk 1]
```diff
@@ -415,6 +415,7 @@ public static MultiLayerNetwork trainModel(
                 .layer( 2, new OutputLayer.Builder( LossFunction.NEGATIVELOGLIKELIHOOD )
                         .activation( Activation.SOFTMAX )
                         .nIn( numInputs ).nOut( outputNum ).build() )
+                .backpropType( BackpropType.Standard )
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/socrates/Socrates.java:418]: Does order matter? Is it ok for `backpropType()` to come after the `layer()`s?

Comment (makosblade) [reply src/main/java/com/openlattice/socrates/Socrates.java:418]: Order does matter in that it appears to no longer be able to go anywhere except for after layer()s

Comment (UnsungHero97) [reply src/main/java/com/openlattice/socrates/Socrates.java:418]: 🤔

**Review by UnsungHero97 — APPROVED** [review]

**Review by jokedurnez — APPROVED** [review]

File: src/main/java/com/openlattice/socrates/Socrates.java (hunk 1 above)
Comment (pckeyes) [reply src/main/java/com/openlattice/socrates/Socrates.java:418]: i think .backpropType() will be meaningless without .backprop()?

**Comment by makosblade** [comment]
Piper found the correct API changes for this. And it turns out `.backprop( boolean )` was deprecated since beta3:
https://deeplearning4j.org/release-notes#deeplearning4j-api-changes-transition-guide-100-beta2-to-100-beta3
Deprecation PR:
https://github.com/eclipse/deeplearning4j/pull/6296/files

I don't think I have enough context/familiarity with the code to update this at the moment but I can take a look later on. Socrates is broken locally until we either roll back to beta3 or update it for these new API changes

**Pull request closed without merging**